### PR TITLE
Add utility methods to rank annotations

### PR DESCRIPTION
**Pull request by robinschmid**
Added methods to CompoundAnnotationUtils to map and rank the best annotations in different ways.

In LoadingsProvider the getAnnotationTypesByPriority and rankUniqueAnnotationTypes now avoids indexOf lookups and can get the best annotaiton type directly from the map.

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/datamodel/features/FeatureAnnotationPriority.java [hunk 1]
```diff
@@ -43,8 +46,8 @@
  * {@link FeatureAnnotationIterator}
  */
 public enum FeatureAnnotationPriority {
-  MANUAL(ManualAnnotationType.class), LIPID(LipidMatchListType.class), SPECTRAL_LIBRARY(
-      SpectralLibraryMatchesType.class), EXACT_COMPOUND(CompoundDatabaseMatchesType.class), FORMULA(
+  MANUAL(ManualAnnotationType.class), SPECTRAL_LIBRARY(
+      SpectralLibraryMatchesType.class), LIPID(LipidMatchListType.class), EXACT_COMPOUND(CompoundDatabaseMatchesType.class), FORMULA(
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/datamodel/features/FeatureAnnotationPriority.java:50]: Ill revert this change, lipid matches > speclib for lipids

**Review by SteffenHeu — COMMENTED** [review]
looks good so far

File: mzmine-community/src/main/java/io/github/mzmine/datamodel/features/FeatureAnnotationPriority.java (hunk 1 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/datamodel/features/FeatureAnnotationPriority.java:50]: :D oh no.
Cannot even imagine lipids should be higher than spectral library.
